content/docs/rfcs/32/README.md

```diff
@@ -0,0 +1,457 @@
+An example for social stake is using InterRep as a registry - 
+users need to prove that they have a highly reputable social media account.
+
+The user's identity is composed of:
+
+```
+{
+    identity_secret: secret_component_arr,
+    identity_secret_hash: poseidonHash(identity_secret),
+    identity_commitment: poseidonHash([identity_secret_hash])
+}
+```
+
+For registration, the user needs to submit their `identity_commitment` (along with any additional registration requirements) to the registry party. 
+Upon registration, they should receive `leaf_index` value which represents their position in the merkle tree. 
+Receiving a `leaf_index` is not a hard requirement and is application specific. 
+The other way around is the users calculating the `leaf_index` themselves upon successful registration.
+
+## Signaling
+
+After registration, 
+the users can participate in the application by sending signals to the other participants in a decentralised manner or to a centralised server. 
+Along with their signal, 
+they need to generate a ZK-Proof by using the circuit with the specification described above. 
+
+For generating a proof, 
+the users need to obtain the required parameters or compute them themselves, 
+depending on the application implementation and client libraries supported by the application.
+For example the users can store the memebrship merkle tree on their end and 
+generate a merkle proof whenever they want to generate a signal. 
+
+### Signal hash
+
+The signal hash is generated by hashing the raw signal (or content) using the `keccak256` hash function, 
+converting the hashed value to number and padding it:
+
+```
+  signal: string = "signal"
+  converted: string = toHex(toUtf8Bytes(signal))
+  signal_hash: bigint = BigInt(keccak256(converted)) >> BigInt(8)
+```
+
+
+### Epoch
+
+The epoch (or external nullifier) is also obtained by hasing a raw string value using `keccak256`:
+
+```
+    raw_epoch: string = "1234"
+    converted: string = keccak256(raw_epoch)
+    hex_string: string = converted.slice(8, 64)
+    epoch: string = "0x" + hex_string.padStart(64, "0")
+```
+
+
+### Merkle proof
+
+The merkle proof should be obtained locally or from a trusted third party. 
+By using the incremental merkle tree algorithm, 
+the merkle can be obtained by providing the `leaf_index` of the `identity_commitment`. 
+The proof (`merkle_proof`) is composed of the following fields:
+
+```
+{
+  root: bigint
+  indices: number[]
+  path_elements: bigint[][]
+}
+```
+
+
+### Generating proof
+
+For proof generation, 
+the user need to submit the following fields to the circuit:
+
+```
+    {
+      identity_secret: identity_secret_hash,
+      path_elements: merkle_proof.path_elements,
+      identity_path_index: merkle_proof.indices,
+      x: signal_hash,
+      epoch: epoch,
+      rln_identifier: rlnIdentifier
+    }
+```
+
+The generated proof (`zk_proof`) is composed of the following fields:
```

Comment:
Elaborate on what these fields stand for (or link to reference that explains)

Reply:
Will do